spec/features/batches_spec.rb

```diff
@@ -522,4 +525,150 @@
+        expect(page).to have_unchecked_field("batch_form_single_barcode")
+
+      end
+
+      it "shows the correct favourite status of the consumable type" do
+        @fav = create(:favourite, user_id: test_user.id, consumable_type_id: @consumable_type1.id)
+        visit edit_batch_path(@batch)
+        expect(page).to have_select("batch_form_consumable_type_id", selected: @consumable_type1.name)
+        expect(page).to have_selector("i.fa.fa-star.fa-3x.favourite")
+      end
+
+      it "shows 'Save Changes' on the submit button" do
+        visit edit_batch_path(@batch)
+        expect(page).to have_selector("input[type=submit][value='Save Changes']")
+      end
+
+      it "shows the title of the batch at the top of the form" do
+        visit edit_batch_path(@batch)
+        expect(page).to have_css("h2", text: 'Editing Batch: ' + @batch.number)
+      end
+    end
+
+    context "submitting the form" do
+      it "updates the appropriate records" do
+        @external_team = create(:supplier)
+
+        visit edit_batch_path(@batch)
+        # consumable type dropdown
+        select @consumable_type2.name, from: "batch_form_consumable_type_id"
+
+        # ingredients form
+        click_button("Add Ingredient")
+        select @consumable_type2.name, from: "batch_form_ingredients__consumable_type_id"
+        select @external_team.name, from: "batch_form_ingredients__kitchen_id"
+
+        # rest of form
+        fill_in "batch_form_expiry_date", with: "11/11/2021"
+        fill_in "batch_form_aliquots", with: "74"
+        fill_in "batch_form_aliquot_volume", with: "9"
+        check("batch_form_single_barcode")
+        click_button "Save Changes"
+
+        visit batch_path(@batch)
+        expect(page).to have_text("Consumable type: " + @consumable_type2.name)
+        expect(page).to have_text("Expiry date: 11/11/2021")
+        expect(page).to have_text("Consumables: 74")
+        expect(page).to have_text("Aliquot Volume: 9mL")
+        expect(page).to have_text("Barcode Type: One per batch")
+        expect(page).to have_text(@external_team.name)
+      end
+
+      it "shows a message upon successful update" do
+        visit edit_batch_path(@batch)
+        click_button("Save Changes")
+        expect(page).to have_text("Reagent batch successfully updated!")
+      end
+    end
+
+    context "trying to edit a printed batch" do
+      it "shows an error if you try to access the URL" do
+        @batch.update(editable: false)
+        visit edit_batch_path(@batch)
+        expect(page).to have_text("This batch has already been printed, so can't be modified.")
+      end
+
+      it "disables the 'Edit Batch' button when viewing a batch" do
+        @batch.update(editable: false)
+        visit batch_path(@batch)
+        expect(page).to_not have_button("Edit Batch")
+      end
+
+      it "doesn't show the corresponding pencil when viewing the batch index" do
```

Comment:
This test is good in that it tests what we want, BUT it's a little too brittle as it is relying on the order of columns in a table (tests involving UIs are always brittle to some extend but we can try to mitigate this).

If you could find a way to find the row in the table of a particular editable batch, and check the editable pencil appears *somewhere in that row*, this would protect us from this test breaking too easily in the future.